pages/notification/NotificationHandler.inc.php

```diff
@@ -68,6 +68,62 @@ function fetchNotification($args, $request) {
 		return $json;
 	}
 
+	/**
+	 * Notification Unsubscribe handler
+	 * @param $args array
+	 * @param $request Request
+	 */
+	function unsubscribe($args, $request) {
+		$validationToken = $request->getUserVar('validate');
+		$notificationId = $request->getUserVar('id');
+
+		if ($validationToken == null || $notificationId == null) {
+			$dispatcher = $this->getDispatcher();
+			$dispatcher->handle404();
```

Comment:
Let's chain these statements together, here and below:

```
$this->getDispatcher()->handle404();
```

Reply:
OK